languages/csharp/exercises/concept/interfaces/.docs/instructions.md

```diff
@@ -0,0 +1,43 @@
+In this exercise you will be doing some more work on remote control cars.
+
+An experimental car has been developed and the test track needs to be adapted to handle both production and experimental models. The two types of car have already been built and you need to find a way to deal with them both on the test track.
+
+In addition, production cars are beginning to have some success. The team boss is keen to maintain the competitive spirit by publishing a ranking of the production cars.
+
+## 1. Enable cars to be driven on the same test track
+
+Please add a method to the `IRemoteControlCar` interface to expose the implementations of `Drive()` for the two types of car.
+
+```csharp
+TestTrack.Race(new ProductionRemoteCcontrolCar());
+TestTrack.Race(new ExperimentalRemoteCcontrolCar());
+// this should execute without an exception being thrown
+```
+
+## 2. Enable the distance travelled by different models on the test track to be compared
+
+Please add a property to the `IRemoteControlCar` interface to expose the implementations of the `DistanceTravelled` property for the two types of car.
+
+```csharp
+var prod = new ProductionRemoteCcontrolCar();
+TestTrack.Race(prod);
+var exp = new ExperimentalRemoteCcontrolCar();
+TestTrack.Race(exp);
+prod.DistanceTravelled
+// => 10
+exp.DistanceTravelled
+// => 20
+```
+
+## 3. Allow the production cars to be ranked
+
+Please implement the `IComparable<T>` interface in the `ProductionRemoteControlCar` class.
```

Comment:
This is part of the task, but not the whole task as the student also needs to rank the cars. Maybe it is worth explaining the ranking bit too. As an alternative, instead of having the student ranking the cars themselves, we _could_ have this task only be about implementing the `IComparable<T>` interface on the `ProductionRemoteControlCar` class, as we could easily then call `.Sort()` on a list of our own choosing to verify the correct implementation of the interface.

Reply:
I have added a test that sorts cars using the default comparison.  This should ensure that students implement `IComparable<T>` rather than using an `IComparer<T>` or even LINQ (the rascals).

I have left the ordering task with improved instructions. I think it is important that students see the connection between `IComparable<T>` and its usage.